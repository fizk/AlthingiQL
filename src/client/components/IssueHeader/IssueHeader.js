import React from 'react';
import PropTypes from 'prop-types';
import {H1, H2, H3} from "../../elements/Headline/index";
import {Row, Column} from "../../elements/Grid/index";

export default class AssemblyHeader extends React.Component {
    static propTypes = {
        issue: PropTypes.shape({
            id: PropTypes.number,
            assembly: PropTypes.shape({
                id: PropTypes.number,
            }),
            category: PropTypes.string,
            name: PropTypes.string,
            subName: PropTypes.string,
            type: PropTypes.string,
            typeName: PropTypes.string,
            typeSubName: PropTypes.string,
            status: PropTypes.string,
            question: PropTypes.string,
            goal: PropTypes.string,
            date: PropTypes.string,
        }),
        loading: PropTypes.bool,
    };

    static defaultProps = {
        issue: {
            id: undefined,
            assembly: {
                id: undefined,
            },
            category: undefined,
            name: undefined,
            subName: undefined,
            type: undefined,
            typeName: undefined,
            typeSubName: undefined,
            status: undefined,
            question: undefined,
            goal: undefined,
            date: undefined,
        },
        loading: false,
    };

    render() {
        return (
            <Row>
                <Column>
                    <H1>{this.props.issue.id} {this.props.issue.name}</H1>
                    <H3>{this.props.issue.typeName}</H3>
                    <H2>{this.props.issue.status}</H2>
                    <time>
                        {this.props.issue.date}
                    </time>
                </Column>
            </Row>
        )
    }
}

