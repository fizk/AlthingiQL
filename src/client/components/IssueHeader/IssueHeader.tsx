import React from 'react';
import PropTypes from 'prop-types';
import {H1, H2, H3} from '../../elements/Headline';
import {Row, Column} from '../../elements/Grid';

export default class IssueHeader extends React.Component {
    static propTypes = {
        issue: PropTypes.shape({
            id: PropTypes.number,
            name: PropTypes.string,
            typeName: PropTypes.string,
            status: PropTypes.string,
            date: PropTypes.string,
        }),
        loading: PropTypes.bool,
    };

    static defaultProps = {
        issue: {
            id: undefined,
            name: undefined,
            typeName: undefined,
            status: undefined,
            date: undefined,
        },
        loading: false,
    };

    render() {
        return (
            <Row>
                <Column>
                    <H1>{this.props.issue.id} {this.props.issue.name}</H1>
                    <H3>{this.props.issue.typeName}</H3>
                    <H2>{this.props.issue.status}</H2>
                    <time>
                        {this.props.issue.date}
                    </time>
                </Column>
            </Row>
        )
    }
}

