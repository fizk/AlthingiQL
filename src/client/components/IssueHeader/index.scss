@import "../../config";

.issue-header {
    display: flex;
}

.issue-header__title {
    padding-right: map_get($space, md);
}

.issue-header__state {

}
