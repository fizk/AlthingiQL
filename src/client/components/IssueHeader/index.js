import React from 'react';
import {graphql, compose, gql} from 'react-apollo';
import IssueHeader from './IssueHeader';

const assemblyQuery = gql`
    query assemblyIssue($assembly: Int! $issue: Int!) {
        AssemblyIssue (assembly: $assembly issue: $issue) {
            id
            assembly {id}
            category
            name
            subName
            type
            typeName
            typeSubName
            status
            question
            goal
            date
        }
    }
`;

export default compose(
    graphql(assemblyQuery, {
        props: all => ({
            issue: all.data.loading === false ? all.data.AssemblyIssue : undefined,
            loading: all.data.loading
        }),
        options: ({assembly, issue}) => ({
            variables: {
                assembly: assembly,
                issue: issue,
            }
        })
    })
)(IssueHeader)
