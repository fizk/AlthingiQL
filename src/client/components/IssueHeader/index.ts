import {graphql, compose, gql} from 'react-apollo';
import IssueHeader from './IssueHeader';

type AssemblyIssue = {
    id: number
    assembly?: {id: number}
    category?: string
    name: string
    subName?: string
    type?: string
    typeName: string
    typeSubName?: string
    status: string
    question?: string
    goal?: string
    date: string
}

const assemblyQuery = gql`
    query assemblyIssue($assembly: Int! $issue: Int!) {
        AssemblyIssue (assembly: $assembly issue: $issue) {
            id
            assembly {id}
            category
            name
            subName
            type
            typeName
            typeSubName
            status
            question
            goal
            date
        }
    }
`;

export default compose<any>( //@todo `any`
    graphql(assemblyQuery, {
        props: (all: {data?: {loading: boolean, AssemblyIssue: AssemblyIssue}}) => ({
            issue: all.data.loading === false ? all.data.AssemblyIssue : undefined,
            loading: all.data.loading
        }),
        options: ({assembly, issue}: {assembly: number, issue: number}) => ({
            variables: {
                assembly: assembly,
                issue: issue,
            }
        })
    })
)(IssueHeader)
