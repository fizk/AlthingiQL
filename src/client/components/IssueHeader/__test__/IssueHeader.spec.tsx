import React from 'react';
import IssueHeader from '../IssueHeader';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

describe('<IssueHeader />', () => {
    test('default props', () => {
        const tree = renderer.create(
            <Router >
                <IssueHeader />
            </Router>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('data', () => {
        const data = {
            issue: {
                id: 1,
                name: 'name',
                typeName: 'type name',
                status: 'status',
                date: '2001-01-01',
            },
            loading: false,
        };
        const tree = renderer.create(
            <Router >
                <IssueHeader {...data} />
            </Router>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});


