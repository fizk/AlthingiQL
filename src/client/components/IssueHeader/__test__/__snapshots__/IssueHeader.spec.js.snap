// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueHeader /> data 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <h1
      className="headline-1"
    >
      1
       
      name
    </h1>
    <h3
      className="headline-3"
    >
      type name
    </h3>
    <h2
      className="headline-2"
    >
      status
    </h2>
    <time>
      2001-01-01
    </time>
  </div>
</div>
`;

exports[`<IssueHeader /> default props 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <h1
      className="headline-1"
    >
       
    </h1>
    <h3
      className="headline-3"
    />
    <h2
      className="headline-2"
    />
    <time />
  </div>
</div>
`;
